src/app/containers/BorrowHistory/index.tsx

```diff
@@ -1,164 +1,76 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useTable, useSortBy } from 'react-table';
-import { Icon, Text } from '@blueprintjs/core';
 import { translations } from 'locales/i18n';
+import { AssetsDictionary } from 'utils/dictionaries/assets-dictionary';
+import { DisplayDate } from 'app/components/ActiveUserLoanContainer/components/DisplayDate';
+import { useGetBorrowHistoryData } from './hooks/useGetBorrowHistoryData';
+import { Table } from 'app/components/Table';
+import { AssetRenderer } from 'app/components/AssetRenderer';
+import { toAssetNumberFormat } from 'utils/display-text/format';
 
-import { BorrowAmount } from '../../components/ActiveBorrowTable/BorrowAmount';
-import { AssetsDictionary } from '../../../utils/dictionaries/assets-dictionary';
-import { CollateralAmount } from '../../components/ActiveBorrowTable/CollateralAmount';
-import { DisplayDate } from '../../components/ActiveUserLoanContainer/components/DisplayDate';
-import { useGetContractPastEvents } from '../../hooks/useGetContractPastEvents';
-import { weiToFixed } from '../../../utils/blockchain/math-helpers';
-import { SkeletonRow } from '../../components/Skeleton/SkeletonRow';
-
-interface Props {}
-
-export function BorrowHistory(props: Props) {
+export const BorrowHistory: React.FC = () => {
   const { t } = useTranslation();
+  const { data, loading } = useGetBorrowHistoryData();
 
-  const { events, loading } = useGetContractPastEvents(
-    'sovrynProtocol',
-    'Borrow',
-  );
+  const rows = useMemo(() => {
+    if (!data || loading) return null;
+    return data.user.borrows.map(item => {
```

Comment:
Could use borrows directly when skipping parent relation to user:

```suggestion
    return data.borrows.map(item => {
```

Reply:
related to https://github.com/DistributedCollective/Sovryn-frontend/pull/2251#discussion_r894343210

Reply:
done